File: question4.c
// Move All Zeroes to End

#include <stdio.h>

void moveZerosToEnd(int arr[], int n) {
    int j = 0; 

    for (int i = 0; i < n; i++) {
        if (arr[i] != 0) {
            arr[j++] = arr[i];
        }
    }

    
    while (j < n) {
        arr[j++] = 0;
    }
}

int main() {
    int arr[] = {1, 2, 0, 4, 3, 0, 5, 0};
    int n = sizeof(arr) / sizeof(arr[0]);

    moveZerosToEnd(arr, n);

    printf("Output: ");
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
}

File: question4.py
# Move All Zeroes to End

def move_zeros_to_end(arr):
    j = 0  

    for i in range(len(arr)):
        if arr[i] != 0:
            arr[j] = arr[i]
            j += 1

    while j < len(arr):
        arr[j] = 0
        j += 1

arr = [1, 2, 0, 4, 3, 0, 5, 0]
move_zeros_to_end(arr)
print("Output:", arr)
